dataplane/standalone/saiserver/attrmgr/attrmgr.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package attrmgr contains an SAI attribute key/value store.
+package attrmgr
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protowire"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+
+	saipb "github.com/openconfig/lemming/dataplane/standalone/proto"
+)
+
+// AttrMgr stores and retrieve the SAI attributes.
+type AttrMgr struct {
+	attrsMu sync.RWMutex
+	attrs   map[string]map[int32]protoreflect.Value
```

Comment:
I think it would be helpful for readers if you explained what the string and the int32 were in the map.

Reply:
done